### DISCUSSION TITLE
Send rtmp stream with OBS to Liquidsoap

### DISCUSSION THREAD

**Discussion by sorintodys** [savonet/liquidsoap#2396] [Q&A]
Hello,

Is it possible to send a stream with obs to liquidsoap and it reacts like do for audio, to replace the stream?

**Comment by toots** [comment]
Hi! Yes, definitely. Liquidsoap can receive `rtmp` stream via `input.rtmp` and `srt` streams via `input.srt`.

**Reply by sorintodys** [reply]
I try this in liq conf:

live = input.rtmp('rtmp://pstnet6.shoutcastnet.com:6106/live/stream')
mediaplaylist = fallback(track_sensitive=false,[live,mediaplaylist])

In OBS i put:

Server: rtmp://pstnet6.shoutcastnet.com:6106/live
Stream Key: stream

But liquidsoap not let OBS to connect.

Liq is start ok and play a folder with videos.

**Reply by sorintodys** [reply]
The entire script is:

```
settings.init.allow_root.set(true)
settings.protocol.ffmpeg.path.set('/var/www/video/encoders/ffmpeg')
audio.channels.set(2)
audio.samplerate.set(48000)
video.frame.width.set(1280)
video.frame.height.set(720)
settings.frame.video.framerate.set(24)
settings.log.file.path.set('/var/www/video/vhosts/6100/log/autodj.log')

mediaplaylist = mksafe(playlist(mode='normal',reload=1,reload_mode='rounds','/var/www/video/vhosts/6100/media/'))

live = input.rtmp(listen=false,'rtmp://pstnet6.shoutcastnet.com:6106/live/stream')

mediaplaylist = fallback(track_sensitive=false,[live,mediaplaylist])

encoder = %ffmpeg(%video(codec='libx264', b='1792k', preset='superfast', tune='zerolatency'), %audio(codec='aac', b='192k'), format='flv')

output.url(url='rtmp://pstnet6.shoutcastnet.com:6100/live/stream', encoder, mediaplaylist)
```

In OBS i put:

Server: rtmp://pstnet6.shoutcastnet.com:6106/live
Stream Key: stream

**Reply by toots** [reply]
Sorry for the delayed response. I am not sure that this is how this works. You need a direct connection between liquidsoap and OBS so, for instance, liquidsoap should be the server and OBS connects to it. This should be the most common situation. `input.rtmp` has a `listen` parameter that turns it into a receiving RTMP server.

**Comment by sorintitu** [comment]
Thank you for answer.
Yes, we need liquidsoap to become rtmp server, is this posible?
 And if is how to set it?

**Reply by toots** [reply]
You need to make sure that `liquidsoap` has a public-facing IP and then I think you can use this:
```
live = input.rtmp(listen=true, 'rtmp://0.0.0.0:6106/live/stream')
```

**Reply by sorintitu** [reply]
This working great !!! It is amaizing neversee the best and more :)))
Thank you very much liquidsoap team for a super application!

**Reply by toots** [reply]
Glad to hear, all the best with your project!
